plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/completion/contributors/FirSuperMemberCompletionContributor.kt

```diff
@@ -0,0 +1,94 @@
+import org.jetbrains.kotlin.idea.completion.ItemPriority
+import org.jetbrains.kotlin.idea.completion.checkers.CompletionVisibilityChecker
+import org.jetbrains.kotlin.idea.completion.context.FirBasicCompletionContext
+import org.jetbrains.kotlin.idea.completion.context.FirNameReferencePositionContext
+import org.jetbrains.kotlin.idea.completion.contributors.helpers.collectNonExtensions
+import org.jetbrains.kotlin.idea.completion.lookups.CallableInsertionOptions
+import org.jetbrains.kotlin.idea.completion.lookups.CallableInsertionStrategy
+import org.jetbrains.kotlin.idea.completion.lookups.detectImportStrategy
+import org.jetbrains.kotlin.idea.frontend.api.KtAnalysisSession
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtCallableSymbol
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtFunctionSymbol
+import org.jetbrains.kotlin.idea.frontend.api.types.KtType
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtSuperExpression
+
+internal class FirSuperMemberCompletionContributor(
+    basicContext: FirBasicCompletionContext,
+    priority: Int
+) : FirCompletionContributorBase<FirNameReferencePositionContext>(basicContext, priority) {
+    override fun KtAnalysisSession.complete(positionContext: FirNameReferencePositionContext) = with(positionContext) {
+        val superReceiver = explicitReceiver as? KtSuperExpression ?: return
+        val expectedType = nameExpression.getExpectedType()
+        val visibilityChecker = CompletionVisibilityChecker.create(basicContext, positionContext)
+        val possibleReceiverScope = superReceiver.getKtType()?.getTypeScope() ?: return
+        val nonExtensionMembers = collectNonExtensions(possibleReceiverScope, visibilityChecker, scopeNameFilter)
+        collectDelegateCallToSuperMember(superReceiver, nonExtensionMembers, expectedType)
+    }
+
+    private fun KtAnalysisSession.collectDelegateCallToSuperMember(
+        superReceiver: KtSuperExpression,
+        nonExtensionMembers: Sequence<KtCallableSymbol>,
+        expectedType: KtType?
+    ) {
+        // A map that contains all containing functions as values, each of which is indexed by symbols it overrides. For example, consider
+        // the following code
+        // ```
+        // class A : Runnable {
+        //   override fun run() {
+        //     val o = object: Callable<String> {
+        //       override fun call(): String {
+        //         super.<caret>
+        //       }
+        //     }
+        //   }
+        // }
+        // ```
+        // The map would contain
+        //
+        // * Runnable.run -> A.run
+        // * Callable.call -> <anonymous object>.call
+        val superFunctionToContainingFunction = superReceiver
+            .parentsOfType<KtNamedFunction>(withSelf = false)
+            .flatMap { containingFunction ->
+                val original = basicContext.toFromOriginalFileMapper.toOriginalFile(containingFunction) ?: return@flatMap emptyList()
```

Comment:
Could you please explain why we need original function here? What will happen if we would use `containingFunction` instead?

Reply:
Using the fake element causes `getAllOverriddenSymbols` to return nothing. Is this unexpected?

Reply:
It should work for the fake element too

Reply:
It just looks like using `toFromOriginalFileMapper` hides the real problem here and in https://github.com/JetBrains/intellij-community/pull/1688

Reply:
OK, it looks like with the fake element, a distinct FirDeclaration is created, which causes the query at [here](https://kotlin.jetbrains.space/p/kotlin/repositories/kotlin/files/51fc2e453fe1657874c2a46a16cb400d64dbed16/compiler/fir/tree/src/org/jetbrains/kotlin/fir/scopes/FirTypeScope.kt?tab=source&line=71&annotate=true) to fail because the identity of the FirElement does not match the one in directOverriddenMap, which was created from the original file.

Can you educate me what's the intended behavior here? Should a fake PsiElement create a distinct "fake" FirElement or should it recover the original FirElement?
* if the former is intended, then it means in the same FIR session, we allow "duplicate" declarations, and we are missing a mechanism to consider them "identical" for the purpose of getting overridden symbols
* if the latter is the intended, then it appears we probably want to get the corresponding original PSI element first and use that to find proper FirElement.

Or am I totally missing the point and it's not intended to work in either of the above mentioned way?

Reply:
As discussed offline, I have removed the mapper and updated LL API to always query with real FIR element in https://kotlin.jetbrains.space/p/kotlin/reviews/217/timeline